Tests/SwiftParserTest/DeclarationTests.swift

```diff
@@ -951,6 +951,41 @@ final class DeclarationTests: ParserTestCase {
     )
   }
 
+  func testTypedThrowsPlacedAfterArrow() {
+    assertParse(
+      "func test() -> 1️⃣throws (any Error) Int",
+      diagnostics: [
+        DiagnosticSpec(
+          message: "'throws (any Error)' must precede '->'",
+          fixIts: ["move 'throws (any Error)' in front of '->'"]
+        )
+      ],
+      fixedSource: "func test() throws(any Error) ->  Int"
+    )
+
+    assertParse(
+      "func test() -> 1️⃣throws (any Error Int",
+      diagnostics: [
+        DiagnosticSpec(
+          message: "'throws (any Error' must precede '->'",
+          fixIts: ["move 'throws (any Error' in front of '->'"]
+        )
+      ],
+      fixedSource: "func test() throws(any Error -> Int"
+    )
+
+    assertParse(
+      "func test() -> 1️⃣throws /* comment */ (any Error) Int",
+      diagnostics: [
+        DiagnosticSpec(
+          message: "'throws /* comment */ (any Error)' must precede '->'",
+          fixIts: ["move 'throws /* comment */ (any Error)' in front of '->'"]
+        )
+      ],
+      fixedSource: "func test() throws(any Error) -> /* comment */  Int"
+    )
```

Comment:
@ahoppen, you've mentioned we should only continue parsing the type if there are no whitespace and comment trivia between `throws` and `.leftParen`. Here I noticed that the comment trivia is left in place in the fixed source after applying the fix-it even though `/* comment */` is included in the message and fix-it. Is it because of the issue of `parsedToken` vs `materializedToken`?

However, `throws /* comment */ (any Error)` is also a legit throws clause, shouldn't it be supported by this fix-it?

Reply:
Sorry, I should have elaborate when I suggested the no spacing rule: If a user writes `func foo() -> throws (Int, Int)`, the `(Int, Int)` very likely isn’t the thrown error type but the return type and I want to avoid generating a Fix-It that moves `throws (Int, Int)` in front of the arrow.

And if a user has a space between `throws` and the thrown error type and also writes the clause after the arrow, I am fine with producing a sub-optimal diagnostic because those two choices/mistakes combined are quite unlikely. 

---

And also congrats for noticing the `throws /* comment */ (any Error)` case. Shows a great attention to detail 🤩